pyne/mcnp.py

```diff
@@ -1990,210 +1987,235 @@ def _read_meshtal_head(self, f):
+        conversion factors are being used.
 
-    def _read_meshtally_head(self, f):
-        """Get the particle type, spacial and energy bounds, and whether or
-        not flux-to-dose conversion factors are being used.
+        Parameters
+        ----------
+        f : str or filestream
+            Filestream of the meshtal file.
+    
+        Returns
+        -------
+        particle : str
+            The particle type, 'neutron' or 'photon'.
+        dose_response : bool
+            True : if this meshtally is modified by a dose function.
+            False : if this meshtally is not modified by a dose function.
         """
+
         line = f.readline()
         if ('neutron' in line):
-            self.particle = 'neutron'
+            particle = 'neutron'
         elif ('photon' in line):
-            self.particle = 'photon'
+            particle = 'photon'
 
         # determine if meshtally flux-to-dose conversion factors are being used.
         line = f.readline()
         dr_str = 'This mesh tally is modified by a dose response function.'
         if line.strip() == dr_str:
-            self.dose_response = True
+            dose_response = True
         else:
-            self.dose_response = False
+            dose_response = False
+
+        return particle, dose_response
+
+    def read_xyze_bounds(self, f):
+        """
+        Read the spatial and energy bounds.
+
+        Parameters
+        ----------
+        f : str or filestream
+            Filestream of the meshtal file.
+    
+        Returns
+        -------
+        x_bounds : list of float
+            Mesh boundaries of X dimension.
+        y_bounds : list of float
+            Mesh boundaries of Y dimension.
+        z_bounds : list of float
+            Mesh boundaries of Z dimension.
+        e_bounds : list of float
+            Energy boundaries.
+        """
 
         # advance the file to the line where x, y, z, bounds start
+        line = f.readline()
         while line.strip() != 'Tally bin boundaries:':
             line = f.readline()
 
-        self.x_bounds = [float(x) for x in f.readline().split()[2:]]
-        self.y_bounds = [float(x) for x in f.readline().split()[2:]]
-        self.z_bounds = [float(x) for x in f.readline().split()[2:]]
+        x_bounds = [float(x) for x in f.readline().split()[2:]]
+        y_bounds = [float(x) for x in f.readline().split()[2:]]
+        z_bounds = [float(x) for x in f.readline().split()[2:]]
         # "Energy bin boundaries" contain one more word than "X boundaries"
-        self.e_bounds = [float(x) for x in f.readline().split()[3:]]
+        e_bounds = [float(x) for x in f.readline().split()[3:]]
+        return x_bounds, y_bounds, z_bounds, e_bounds
 
-        self.dims = [0, 0, 0] + [len(self.x_bounds) - 1,
-                                 len(self.y_bounds) - 1,
-                                 len(self.z_bounds) - 1]
+    def read_column_order(self, f):
```

Comment:
This method doesn't use `self` and can be made a static method.

Reply:
I kept the `self` here, for one thing, there are two `read_xxx` functions that uses `self`. For another thing, all there readers operates the same `meshtal` file. I think it's better to keep them linked to the same object (`self`).

Reply:
If we aren’t using `self` in the method to access or set attributes, it doesn’t change how the information is linked to self or how the file handle, `f` is managed. And that method would still maintain it’s context as a reader by nature of being part of the `Meshtal` class definition.

All that being said, `column_idx` is now one of the class attributes (a good update). So, yes, we should keep `self` here. 😃 

We’ll want to update the docstring of this method though as it still indicates that we’re returning a `dict`.

Reply:
Docstring of this function is update in https://github.com/pyne/pyne/pull/1199/commits/bae3bf1d7e03f507e8729edd3197c8daf20853e5.